src/main/java/org/sefglobal/scholarx/service/ProgramService.java

```diff
@@ -204,4 +205,115 @@ public Mentor applyAsMentor(long programId, long profileId, Mentor mentor)
         mentor.setState(EnrolmentState.PENDING);
         return mentorRepository.save(mentor);
     }
+
+    /**
+     * Retrieves the {@link Mentor} of a user if the user is a mentor
+     *
+     * @param programId which is the id of the {@link Program}
+     * @param profileId which is the id of the {@link Profile}
+     * @return {@link Mentor}
+     *
+     * @throws ResourceNotFoundException if the requesting {@link Program} doesn't exist
+     */
+    public Mentor getLoggedInMentor(long programId, long profileId)
+            throws ResourceNotFoundException {
+        Optional<Mentor> optionalMentor = mentorRepository.findByProfileIdAndProgramId(profileId, programId);
+        if (!optionalMentor.isPresent()) {
+            String msg = "Error, Mentor by profile id: " + profileId + " and program id: " + programId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+        return optionalMentor.get();
+    }
+
+    /**
+     * Update the application and questionnaire of a {@link Mentor}
+     *
+     * @param profileId which is the Profile id of the {@link Mentor} to be updated
+     * @param programId which is the Program id of the {@link Mentor} to be updated
+     * @param mentor    with the application and prerequisites of the mentor to be updated
+     * @return the updated {@link Mentor}
+     *
+     * @throws ResourceNotFoundException is thrown if the mentor doesn't exist
+     */
+    public Mentor updateApplicationAndPrerequisites(long profileId, long programId, Mentor mentor)
+            throws ResourceNotFoundException {
+        Optional<Mentor> optionalMentor = mentorRepository.findByProfileIdAndProgramId(profileId, programId);
```

Comment:
Following validations need to be done to this obj,
1. if application is updated : mentor should be `PENDING`
2. if prerequisites is updated : mentor should be `PENDING` or `APPROVED` and program status should be < `MENTEE_APPLICATION`